test/python/utils/test_deprecation.py

```diff
@@ -15,7 +15,89 @@
 from textwrap import dedent
 
 from qiskit.test import QiskitTestCase
-from qiskit.utils.deprecation import add_deprecation_to_docstring
+from qiskit.utils.deprecation import (
+    add_deprecation_to_docstring,
+    deprecate_function,
+    deprecate_arguments,
+)
+
+
+class TestDeprecationDecorators(QiskitTestCase):
+    """Test that the decorators in ``utils.deprecation`` correctly log warnings and get added to
+    docstring."""
+
+    def test_deprecate_arguments_message(self) -> None:
+        """Test that `@deprecate_arguments` adds the correct message to the docstring."""
+
+        @deprecate_arguments(
+            {"old_arg1": "new_arg1", "old_arg2": None},
+            category=PendingDeprecationWarning,
+            since="9.99",
+        )
+        def my_func() -> None:
+            pass
+
+        self.assertEqual(
+            my_func.__doc__,
+            dedent(
```

Comment:
not sure about removing idents, as they are kinda important in this context. Do you think that a literal comparison is too strong?

Reply:
This still considers indentation. Note how there are two spaces underneath the line `.. deprecated::`. This is how the docstring renders when there was no prior docstring, which is what we want.

Also, for context, these tests are not focused on ensuring that `add_deprecation_to_docstring` handles all the edge cases of indentation. We already have a whole suite of unit tests for that dedicated purpose. These unit tests here are only to check that we:

1. Generate the right message
2. Actually call the helper function `add_deprecation_to_docstring`